GitHub Actions changed to Maven 3.9.0 which breaks the build

We get exceptions such as

```
Error: [ERROR] Internal error: java.lang.NullPointerException: Cannot invoke "org.apache.maven.repository.internal.ModelCacheFactory.createCache(org.eclipse.aether.RepositorySystemSession)" because "this.modelCacheFactory" is null -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.NullPointerException: Cannot invoke "org.apache.maven.repository.internal.ModelCacheFactory.createCache(org.eclipse.aether.RepositorySystemSession)" because "this.modelCacheFactory" is null
```

I expect upgrading to newest Tycho is the solution, which means going from 2.x to 3.x version of Tycho.

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -31,6 +31,10 @@ jobs:
         java-version: '17'
         distribution: 'temurin'
         cache: maven
+    - name: Setup Maven
+      uses: stCarolas/setup-maven@v.4.5
+      with:
+        maven-version: 3.8.7
     - name: Install GCC & GDB & other build essentials
       run: |
         sudo apt-get -y install build-essential gcc g++ gdb gdbserver

--- a/.github/workflows/code-cleanliness.yml
+++ b/.github/workflows/code-cleanliness.yml
@@ -21,6 +21,10 @@ jobs:
         java-version: '17'
         distribution: 'temurin'
         cache: maven
+    - name: Setup Maven
+      uses: stCarolas/setup-maven@v.4.5
+      with:
+        maven-version: 3.8.7
     - name: Install dependencies
       run: |
         sudo apt-get update && sudo apt-get install -y --no-install-recommends \

